### PR TITLE
fix: prevent integer overflow in CDS-to-transcript position conversion

### PR DESCRIPTION
**Pull request by nh13**
## Summary

- Replace unsafe `as u64` casts with `u64::try_from()` at 5 sites in `src/normalize/mod.rs` to prevent integer overflow when normalizing 5' UTR variants with deeply negative CDS positions (e.g., `c.-88`)
- Guard `extract_position_range()` to return `None` for negative CDS bases, skipping simple normalization for UTR variants
- Add 4 tests covering underflow detection, valid UTR5 positions, negative CDS base handling, and deep-negative integration

## Context

Stacked on #5. When normalizing ~57.5M ClinVar HGVS patterns, 1,952 fail due to `cds_to_tx_pos()` computing a negative intermediate value (e.g., `cds_start=44, base=-88` → `44 + (-88) - 1 = -45`) then casting to `u64`, which wraps to ~1.8×10¹⁹.

## Test plan

- [x] `cargo nextest run --features dev` — 2780 tests pass
- [x] `cargo clippy --features dev -- -D warnings` — clean
- [x] `cargo fmt --check` — clean
- [ ] Re-run `ferro normalize` on the 1,952 coordinate-error patterns to confirm clean errors instead of panics/wraps

### CODE REVIEW

**Comment by nh13** [comment]
@coderabbitai review

File: src/normalize/mod.rs [hunk 1]
```diff
@@ -2955,4 +3135,202 @@ mod tests {
...
+        let provider = MockProvider::with_test_data();
+        let normalizer = Normalizer::new(provider);
+        // c.-88 with cds_start=5 → 5 + (-88) - 1 = -84, which would wrap to huge u64
+        let variant = parse_hgvs("NM_001234.1:c.-88A>G").unwrap();
+        let result = normalizer.normalize(&variant);
+        // Should either return an error or handle gracefully — must not panic
+        assert!(
+            result.is_err() || result.is_ok(),
+            "Should not panic on deeply negative 5' UTR position"
+        );
```
[the remark this answers is not in the document]
Comment (nh13) [reply src/normalize/mod.rs]: Fixed in 018f968 — replaced the tautological assertion with `let _ = result` since the test only verifies no panic occurs.

**Comment by nh13** [comment]
Addressed CodeRabbit review feedback in 018f968:

1. **normalize/mod.rs:3193-3196** (actionable) — Replaced tautological `assert!(result.is_err() || result.is_ok())` with `let _ = result`. The test's purpose is only to verify no panic on deeply negative 5' UTR positions.

2. **normalize/mod.rs:1023-1131** (nitpick, duplication with `normalize_intronic_cds`) — Acknowledged but skipping for now. The types differ enough (`CdsPos` vs `TxPos`, different mapper methods) that a clean abstraction would require closures or a trait for coordinate conversions, adding complexity without reducing bug surface. Worth revisiting if a third intronic normalization path is added.

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
Addressed latest CodeRabbit nitpick feedback, squashed to a single commit (6e44e8e):

1. **mod.rs:47-60** (unify offset helpers) — Skipping. `CdsPos` and `TxPos` are intentionally distinct types with no shared trait. Adding a trait for a 3-line helper is over-engineering.

2. **mod.rs:1083-1094** (intron lookup uses only `start_pos`) — Skipping. This is a pre-existing design consistent with biocommons/hgvs. Cross-intron variants are rare edge cases where HGVS normalization semantics are not well-defined.

3. **mod.rs:3278-3332** (test only asserts `is_ok()`) — Fixed. Added assertion that normalized output retains intronic notation (`+` or `-`).
